src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/QueryIT.java

```diff
@@ -1740,6 +1740,32 @@ public void backticksQuotedAliasInJDBCResponseTest() {
         assertTrue(response.contains("\"alias\": \"name\""));
     }
 
+    @Test
+    public void caseWhenSwitchTest() throws IOException {
+        JSONObject response = executeQuery("SELECT CASE age " +
+                "WHEN '30' THEN '1' " +
+                "WHEN '40' THEN 2 " +
+                "ELSE '0' END AS cases FROM " + TEST_INDEX_ACCOUNT + " WHERE age IS NOT NULL");
+        JSONObject hit = getHits(response).getJSONObject(0);
+        String age = hit.query("/_source/age").toString();
+        String cases = age.equals("30") ? "1" : age.equals("40") ? "2" : "0";
+
+        assertThat(cases, equalTo(hit.query("/fields/cases/0")));
+    }
+
+    @Test
+    public void caseWhenJdbcResponseTest() {
```

Comment:
How do the tests behave when ORDER BY/GROUP BY are added?

Reply:
It's gonna be still working well. Thanks for reminding! I'll add them into the query.

Reply:
But I've no idea if it makes any sense in sql. Since the GROUP BY and ORDER BY field should also be in the SELECT field, I didn't figure out any good way to measure the response content. 

How about I creating another test to verify that it works well, since the CASE statement failed to pass the jdbc formatter and threw exceptions before I made this modification, what do you think?